src/main/java/software/amazon/event/kafkaconnector/EventBridgeWriter.java

```diff
@@ -101,6 +110,28 @@ public EventBridgeWriter(EventBridgeSinkConfig config) {
     this.eventBridgeMapper = new DefaultEventBridgeMapper(config);
     this.batching = new DefaultEventBridgeBatching();
 
+    if ((config.offloadingS3defaultBucket != null) && !config.offloadingS3defaultBucket.isEmpty()) {
+      var s3client =
+          S3AsyncClient.builder()
+              .credentialsProvider(credentialsProvider)
+              .endpointOverride(endpointUri)
+              .forcePathStyle(endpointUri != null)
+              .httpClientBuilder(AwsCrtAsyncHttpClient.builder())
+              .overrideConfiguration(clientConfig)
+              .region(Region.of(this.config.region))
+              .build();
+      var bucketName = StringUtils.trim(config.offloadingS3defaultBucket);
+      var jsonPathExp = StringUtils.trim(config.offloadingDefaultFieldRef);
+      var cache = new FifoCache<MessageDigestCacheKey, String>(1000); // TODO
```

Comment:
question: use UUID type instead of string for value? do you see any benefit doing so or is string the better option?

Reply:
Changed to UUID.